webapp/crashstats/libbugzilla.py

```diff
@@ -69,6 +100,35 @@ def frame_generator(thread):
     return frames
 
 
+def java_exception_to_frames(frames):
+    """Build frame information from java_exception stack
+
+    :arg frames: the frames for a stacktrace in a java_exception structure
+
+    :returns: list of frame information dicts with keys "frame", "module", "signature",
+        "source"
+
+    """
+    new_frames = []
+    for i, frame in enumerate(islice(frames, MAX_FRAMES)):
+        source = frame.get("filename") or ""
+        if source and frame.get("lineno") is not None:
+            source = f"{source}:{frame['lineno']}"
+
+        module = truncate(frame.get("module") or "?", 80)
```

Comment:
I looked at some `java_exception` data and some of the modules are hella long, so I added module truncation. In the `java_exception` data I looked at, I didn't find a case where the function signature was too long, but I figured we should keep it just in case.